server/src/main/java/io/deephaven/server/barrage/BarrageStreamGenerator.java

```diff
@@ -419,35 +419,40 @@ private InputStream getInputStream(final View view, final ByteBuffer metadata, f
         final long numRows;
         final int nodesOffset;
         final int buffersOffset;
-        try (final WritableObjectChunk<ChunkInputStreamGenerator.FieldNodeInfo, Values> nodeOffsets =
-                WritableObjectChunk.makeWritableChunk(addColumnData.length);
-                final WritableLongChunk<Values> bufferInfos =
-                        WritableLongChunk.makeWritableChunk(addColumnData.length * 3)) {
-            nodeOffsets.setSize(0);
-            bufferInfos.setSize(0);
+        try (final SizedChunk<Values> nodeOffsets = new SizedChunk<>(ChunkType.Object);
+                final SizedLongChunk<Values> bufferInfos = new SizedLongChunk<>()) {
+            nodeOffsets.ensureCapacity(addColumnData.length);
+            nodeOffsets.get().setSize(0);
+            bufferInfos.ensureCapacity(addColumnData.length * 3);
+            bufferInfos.get().setSize(0);
 
             final MutableLong totalBufferLength = new MutableLong();
             final ChunkInputStreamGenerator.FieldNodeListener fieldNodeListener =
-                    (numElements, nullCount) -> nodeOffsets
-                            .add(new ChunkInputStreamGenerator.FieldNodeInfo(numElements, nullCount));
+                    (numElements, nullCount) -> {
+                        nodeOffsets.ensureCapacityPreserve(nodeOffsets.get().size() + 1);
+                        nodeOffsets.get().asWritableObjectChunk()
+                                .add(new ChunkInputStreamGenerator.FieldNodeInfo(numElements, nullCount));
+                    };
 
             final ChunkInputStreamGenerator.BufferListener bufferListener = (length) -> {
                 totalBufferLength.add(length);
-                bufferInfos.add(length);
+                bufferInfos.ensureCapacityPreserve(bufferInfos.get().size() + 1);
+                bufferInfos.get().add(length);
             };
             numRows = columnVisitor.visit(view, addStream, fieldNodeListener, bufferListener);
 
-            RecordBatch.startNodesVector(header, nodeOffsets.size());
-            for (int i = nodeOffsets.size() - 1; i >= 0; --i) {
-                final ChunkInputStreamGenerator.FieldNodeInfo node = nodeOffsets.get(i);
+            RecordBatch.startNodesVector(header, nodeOffsets.get().size());
+            for (int i = nodeOffsets.get().size() - 1; i >= 0; --i) {
+                final ChunkInputStreamGenerator.FieldNodeInfo node =
```

Comment:
Worth doing the repeated part outside the loop (get() through cast)? Current chunks are fixed after the visit is done.